scripts/mkcloud

```diff
@@ -771,9 +773,54 @@ function testsetup()
     onadmin testsetup
     local ret=$?
     $scp root@$adminip:tempest.log "$artifacts_dir/tempest.log"
+
+    # Register the cloud in Rally and import the results
+    if [ -n "$rally_server" ]; then
+        local title=$(testname)
+        local type=$(testtype)
+
+        $scp root@$adminip:.openrc "$artifacts_dir/.openrc"
+        $scp "$artifacts_dir/tempest.log" rally@$rally_server:/tmp/tempest.$$.log
+        $scp "$artifacts_dir/.openrc" rally@$rally_server:/tmp/.openrc.$$
+        $ssh rally@$rally_server "
+            source rally/bin/activate
+            source /tmp/.openrc.$$
+            deployment=\$(rally deployment create --fromenv --name=\"$title\" | awk '/Using deployment:/{print \$3}')
+            rally verify start $type --tempest-log=/tmp/tempest.$$.log --deployment=\$deployment
+            rm /tmp/tempest.$$.log
+            rm /tmp/.openrc.$$
+        "
+    fi
+
     return $ret
 }
 
+function testname()
+{
+    local name="$$ - $cloudsource/ha:$hacloud"
+
+    # Check if this instance of mkcloud is running inside Jenkins
+    if [ -n "$JENKINGS_URL" ] ; then
+        name="$BUILD_NUMBER - $JOB_NAME - $BUILD_URL"
+    fi
+
+    echo $name
+}
+
+function testtype()
+{
+    # If `tempestopstions` is not set, or contains `smoke`, is a smoke
+    # test, in other case is a full or ad-hoc tesmpest test
+    local type="smoke"
+    if [ -n "$tempestoptions" ]; then
+        if ! [[ $tempestoptions =~ .*smoke.* ]]; then
+            type="full or ad-hoc"
```

Comment:
Can we somehow try a bit harder to be affirmative it's full? For instance, if the options only have `--subunit`, then it's full. The main question is when we have something else in the options, where it might indeed be "ad-hoc".